probe-rs/src/architecture/arm/core/cortex_m.rs

```diff
@@ -131,6 +131,42 @@ impl MemoryMappedRegister for Cpacr {
     const NAME: &'static str = "CPACR";
 }
 
+bitfield! {
+    #[derive(Copy, Clone)]
+    /// IPSR - Interrupt Program Status Register (a variant of xPSR)
+    pub struct Ipsr(u32);
+    impl Debug;
+    pub exception, _: 8,0;
+}
+
+impl Ipsr {
+    /// Returns true if the current exception is one of the Coretex-M core exceptions.
+    pub fn is_core_exception(&self) -> bool {
+        (1..=15).any(|i| self.exception() == i)
+    }
+
+    /// Returns true if the current exception is a ISR.
+    pub fn is_isr(&self) -> bool {
+        self.exception() > 15
+    }
+
+    /// Subtracts the offset of the ISR vector table to get the index of the ISR.
+    pub fn isr_index(&self) -> u32 {
+        self.exception() - 16
+    }
+}
+
+impl TryFrom<RegisterValue> for Ipsr {
+    type Error = crate::Error;
+
+    fn try_from(value: RegisterValue) -> Result<Self, Self::Error> {
+        match value {
+            RegisterValue::U32(value) => Ok(Self(value)),
+            _ => Err(Error::Arm(ArmError::AddressOutOf32BitAddressSpace)),
```

Comment:
This error looks a bit weird. How can a register value be out of the address space?

Reply:
I am open to suggestions. Because `RegisterValue` can by U32/64/128, in theory, it can be out of that space. In reality, the architecture checks done in the code should prevent this error from ever triggering.

I find the `RegisterValue` construct (introduced with the 64-bit support work) quite cumbersome to work with, but I don't have any constructive suggestions for now. Any refactoring ideas for RegisterValue will be welcomed :)

Reply:
To be more specific about my challenge with `RegisterValue` ... the only way you can access the integer value is through the `TryFrom` implementation, it is impossible to use or manipulate the value in closures (like `map` and `into`) and as a result, the code becomes very verbose with `if let` usage.

Reply:
My main gripe was just that the error variant doesn't make any sense. If the error would say that the value is too wide, that would be fine with me.

Reply:
Adjusted :)